sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,268 @@
+            return CLASS_FIELD_HASH_MAP.get(type);
+        }
+        List<Field> childFields = build((Class<?>)type, classChain, logger);
+        Field searchField = convertToBasicSearchField(classField, logger);
+        searchField.setFields(childFields);
+        searchField.setSearchable(false); // TODO
+        searchField.setFilterable(false); // TODO
+        CLASS_FIELD_HASH_MAP.put((Class<?>) type, searchField);
+        return searchField;
+    }
+
+    private static boolean isSupportedNoneParameterizedType(Type type) {
+        return SUPPORTED_NONE_PARAMETERIZED_TYPE.containsKey(type);
+    }
+
+    private static Field buildNoneParameterizedType(java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        Field searchField = convertToBasicSearchField(classField, logger);
+        return enrichWithAnnotation(searchField, classField, logger);
+    }
+
+
+    private static boolean isArrayOrList(Type type) {
+        if (type.getClass().isArray()) {
+            return true;
+        }
+        return isList(type);
+    }
+
+    private static boolean isList(Type type) {
+        if (type instanceof ParameterizedType) {
+            Type rawType = ((ParameterizedType) type).getRawType();
+            return List.class.isAssignableFrom((Class<?>) rawType);
+        }
+        return false;
+    }
+
+    private static Field buildCollectionField(java.lang.reflect.Field classField,
+        Stack<Class<?>> classChain, ClientLogger logger) {
+        Type componentOrElementType = getComponentOrElementType(classField.getGenericType(), logger);
+        if (isSupportedNoneParameterizedType(componentOrElementType)) {
+            Field searchField = convertToBasicSearchField(classField, logger);
+            return enrichWithAnnotation(searchField, classField, logger);
+        }
+        List<Field> childFields = build((Class<?>)componentOrElementType, classChain, logger);
+        Field searchField = convertToBasicSearchField(classField, logger);
+        searchField.setFields(childFields);
+        return searchField;
+    }
+
+    private static Type getComponentOrElementType(Type arrayOrListType, ClientLogger logger) {
+        if (arrayOrListType.getClass().isArray()) {
+            return arrayOrListType.getClass().getComponentType();
+        }
+        if (isList(arrayOrListType)) {
+            ParameterizedType pt = (ParameterizedType) arrayOrListType;
+            return pt.getActualTypeArguments()[0];
+        }
+        throw logger.logExceptionAsError(new RuntimeException("Should not be there"));
+    }
+
+    private static Field convertToBasicSearchField(java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        Field searchField = new Field();
+        searchField.setName(classField.getName());
+        DataType dataType = covertToDataType(classField.getGenericType(), false, logger);
+        searchField.setType(dataType);
+        searchField.setKey(false);
```

Comment:
If these all default to `false`, why set them just to potentially override them later with `enrichWithAnnotations` ?

Reply:
This is good for some field without annotation. 
e.g.
```
private String id;
```

We convert this to Field with Field (name=id, type=EDM.String, isSortable=true, etc)
This is align with the one with annotation.